internal/armada/server/submit.go

```diff
@@ -529,6 +534,16 @@ func (server *SubmitServer) createJobs(request *api.JobSubmitRequest, owner stri
 	return jobs, nil
 }
 
+func enrichText(text map[string]string, jobId string) map[string]string {
+	returnText := util.DeepCopy(text)
```

Comment:
do we need to deep copy? can we not just replace in-place?

Reply:
we probably could. Something about making non mutative functions generally feels better to me but I guess this isn't that important or used again?